providers/multiprovider/src/main/java/dev/openfeature/contrib/providers/multiprovider/FirstSuccessfulStrategy.java

```diff
@@ -0,0 +1,38 @@
+package dev.openfeature.contrib.providers.multiprovider;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.FeatureProvider;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * First Successful Strategy.
+ * Similar to “First Match”, except that errors from evaluated providers do not halt execution.
+ * Instead, it will return the first successful result from a provider.
+ * If no provider successfully responds, it will throw an error result.
+ */
+@Slf4j
+public class FirstSuccessfulStrategy extends BaseStrategy {
+
+    public FirstSuccessfulStrategy(Map<String, FeatureProvider> providers) {
+        super(providers);
+    }
+
+    @Override
+    public <T> ProviderEvaluation<T> evaluate(String key, T defaultValue, EvaluationContext ctx,
+              Function<FeatureProvider, ProviderEvaluation<T>> providerFunction) {
+        for (FeatureProvider provider: getProviders().values()) {
+            try {
+                return providerFunction.apply(provider);
```

Comment:
Not all providers throw exceptions sometimes, they return the information in the provider evaluation instead of throwing as exceptions for flows is expensive see https://github.com/open-feature/java-sdk/pull/1095 - applies to all strategies

Reply:
updated accordingly.